### PR TITLE
Advanced usage docs split

### PR DESCRIPTION
**Pull request by zzzirk**
Removed the advanced_usage.rst file after separating the sections within into their own files in the advanced directory per #744 .

### CODE REVIEW

**Comment by hackebrot** [comment]
Hi @zzzirk! 👋 

Thank you for picking this up!

Have you tried running ``$ make docs`` on your local environment?

**Comment by hackebrot** [comment]
@zzzirk On an unrelated note: The email that you are using for your commits seems to not be associated with your GitHub account. Is this intentional?

**Comment by zzzirk** [comment]
Interesting.  I will look into why my email isn't working.  Yes, I have run make docs in my local environment and I verified all the docs pages worked and showed.  Did you try it as well and have issues with my branch?

**Comment by hackebrot** [comment]
No, I haven't. I figured it might help with this:

> This breaks the sections out, but I am not sold on how it looks in the sidebar.

**Comment by zzzirk** [comment]
@hackebrot Gah, I found the email issue.  Making the changes on my work machine and didn't change my email for the cookiecutter clone.  I've fixed it for future commits.

As per the sidebar, I just don't care for how it looks.  It is totally functional, but the way it was with the Advance Usage section all in a single page looked cleaner as far as the sidebar goes.  I like having each of those broken out.  I will see if I can find a nicer happy middle ground.  I'm glad to take any feedback anyone else might have.  I just wanted to help out with easy things I could.

**Comment by hackebrot** [comment]
Ah, I see. Hey your contribution is greatly appreciated! 🙇 

I may have misunderstood your comment earlier, sorry about that.

**Comment by willingc** [comment]
@zzzirk You can change the toctree's maxdepth to 1 and it should clean up the sidebar.

Another option is to create an advanced/index.rst and put a table of contents in there for the entire advanced section.

Nice job! The sidebar could be a follow up PR.

**Comment by zzzirk** [comment]
@willingc I had the thoughts of using maxdepth and an advanced/index.rst.  I personally like the latter as it just feels cleaner to me.  I was planning on playing around with it a bit and seeing if I could come up with something that was more aesthetically pleasing (to me at least).

**Comment by michaeljoseph** [comment]
LGTM 👍

File: docs/advanced/replay.rst [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+In other words, it persists your **input** for a template and fetches it when you run the same template again.
+
+Example for a replay file (which was created via ``cookiecutter gh:hackebrot/cookiedozer``)::
+
+    {
+        "cookiecutter": {
+            "app_class_name": "FooBarApp",
+            "app_title": "Foo Bar",
+            "email": "raphael@hackebrot.de",
```
Comment (willingc) [inline docs/advanced/replay.rst]: Perhaps change email to ``raphael@example.com``.

File: docs/advanced/user_config.rst [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+use the cli option ``--default-config`` instead. Preventing Cookiecutter from loading
+user settings is crucial for writing integration tests in an isolated environment.
+
+Example user config:
+
+.. code-block:: yaml
+
+    default_context:
+        full_name: "Audrey Roy"
+        email: "audreyr@gmail.com"
```
Comment (willingc) [inline docs/advanced/user_config.rst]: Perhaps change email to ``audreyr@example.com``

File: docs/advanced/user_config.rst [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+Possible settings are:
+
+* default_context: A list of key/value pairs that you want injected as context
+  whenever you generate a project with Cookiecutter. These values are treated
+  like the defaults in `cookiecutter.json`, upon generation of any project.
+* cookiecutters_dir: Directory where your cookiecutters are cloned to when you
+  use Cookiecutter with a repo argument.
+* replay_dir: Directory where Cookiecutter dumps context data to, which
+  you can fetch later on when using the `replay feature`_.
```
Comment (willingc) [inline docs/advanced/user_config.rst]: Sphinx warns that `replay feature` does not link to something.

**Comment by willingc** [comment]
Nicely done @zzzirk. I left a few small cleanup items which you may wish to change now or in the future. Thanks!

**Comment by zzzirk** [comment]
I appreciate the suggestions @willingc.  I've made the changes you recommended.

**Comment by michaeljoseph** [comment]
I am 😍 on this- @willingc, please give it a quick 👀 ?

**Comment by willingc** [comment]
@michaeljoseph LGTM.

@zzzirk Thanks. Nicely done :smile:

**Comment by michaeljoseph** [comment]
@zzzirk 👌

**Pull request merged**